src/stream_manager.cc

```diff
@@ -34,15 +34,20 @@
 
 unsigned CUstream_st::sm_next_stream_uid = 0;
 
-// SST memcpy callbacks
-extern void SST_callback_memcpy_H2D_done();
-extern void SST_callback_memcpy_D2H_done();
+// SST memcpy callbacks, called after a stream operation is done via record_next_done()
+extern void SST_callback_memcpy_H2D_done(uint64_t dst, uint64_t src, size_t count, cudaStream_t stream);
+extern void SST_callback_memcpy_D2H_done(uint64_t dst, uint64_t src, size_t count, cudaStream_t stream);
 extern void SST_callback_memcpy_to_symbol_done();
 extern void SST_callback_memcpy_from_symbol_done();
-__attribute__((weak)) void SST_callback_memcpy_H2D_done() {}
-__attribute__((weak)) void SST_callback_memcpy_D2H_done() {}
+extern void SST_callback_cudaEventSynchronize_done(cudaEvent_t event);
+extern void SST_callback_kernel_done(cudaStream_t stream);
+__attribute__((weak)) void SST_callback_memcpy_H2D_done(uint64_t dst, uint64_t src, size_t count, cudaStream_t stream) {}
```

Comment:
Why do we need the sst symbols to be weak? What else is overriding them?

Reply:
These are the callback functions used to sync with SST side that certain stream operations are completed. They are not used by gpgpusim or accelsim. The SST balar will have the actual definitions of these functions.

Reply:
If we dont provide these weak definitions, gpgpusim will be fine, but accelsim will report linktime errors as it is an executable rather a shared object.